persistent/persistence.py

```diff
@@ -95,12 +98,18 @@ def _get_oid(self):
     def _set_oid(self, value):
         if value == self.__oid:
             return
-        if value is not None:
-            if not isinstance(value, OID_TYPE):
-                raise ValueError('Invalid OID type: %s' % value)
-        if self.__jar is not None and self.__oid is not None:
-            raise ValueError('Already assigned an OID by our jar')
-        self.__oid = value
+        # The C implementation allows *any* value to be
+        # used as the _p_oid.
```

Comment:
@mgedmin me too, then I realized that  `persistent.interfaces.IPersistent` documents them:

```python
_p_oid = Attribute(
        """The object id.

        It is up to the data manager to assign this.

        The special value None is reserved to indicate that an object
        id has not been assigned.  Non-None object ids must be non-empty
        strings.  The 8-byte string consisting of 8 NUL bytes
        ('\x00\x00\x00\x00\x00\x00\x00\x00') is reserved to identify the
        database root object.

        Once assigned an OID, an object cannot be re-assigned another.
        """)
```